astrauli/user-posts-code-challenge
# Query posts by user

### Description
- Implement api endpoint `GET /api/users/:id/posts`
- Add tests

### Test plan
- Ensure DB is running locally - `docker-compose up -d`
- Start server - `npm run dev`
- Verify test plan runs without failures - `npm run test`

#### Setup
You can create user posts with these steps:
1. Create user
```
curl --location --request POST 'localhost:3000/api/users' \
--header 'Content-Type: application/json' \
--data-raw '{
    "fullName": "New name",
    "email": "NewEmail@test.com",
    "username": "new_username",
    "dateOfBirth": "2023-08-12"
}'
```
2. Create a user post
(use ID from above request)
```
curl --location --request POST 'localhost:3000/api/posts/user/:id' \
--header 'Content-Type: application/json' \
--data-raw '{
    "title": "title",
    "description": "description"
}'
```
3.  Get posts by user ID
4. (use ID from first request)
```
curl --location --request GET 'localhost:3000/api/users/:id/posts'
```
